dbms/src/AggregateFunctions/ReservoirSampler.h

```diff
@@ -108,9 +108,12 @@ class ReservoirSampler
       */
     double quantileInterpolated(double level)
     {
-        if (samples.empty())
+        if (samples.empty()) {
+            if (DB::IsDecimalNumber<T>) {
```

Comment:
Why do we return 0 only for decimal here? Also just return 0 is fine here, since double is a built-in type.

Reply:
for other types in case with empty samples clickhouse team want return nan for now